data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/Operator.java

```diff
@@ -6,6 +6,8 @@
 package org.opensearch.dataprepper.expression;
 
 interface Operator<T> {
+    Integer getRuleIndex();
```

Comment:
Can you clarify how this index will work in the listener?

Reply:
So @sbayer55 and I figured out a better approach as follows:

In Operator implementation:
* replace getRuleIndex() with the following:
```
...
@Override
  public Boolean shouldEvaluate(final ParserRuleContext ctx) {
      return ctx.getRuleIndex() == DataPrepperExpressionParser.RULE_equalityOperatorExpression;
  }
...
```
Then in the listener, we only need to override exitEveryRule instead of exit each expression:

```
@Override
    public void exitEveryRule(ParserRuleContext ctx) {
        if (!operandStack.isEmpty()) {
            final int operatorSymbol = operatorSymbolStack.peek();
            final Operator<?> op = operatorProvider.getOperator(operatorSymbol);
            if (op.shouldEvaluate(ctx)) {
                performSingleOperation(op.getNumberOfArgs());
            }
        }
    }
```

Reply:
@chenqi0805 , I think this approach makes sense. My only concern is whether `ParserRuleContext` is the right class to pass in. I'm not as familiar with the ANTLR APIs as you, so I'll leave it to you. But, if there are other classes which could be evaluated, consider a more decoupled parameter - perhaps a super-interface, or just passing in the `int`.